### DISCUSSION TITLE
Preserve Dates on Import: Created and Modified dates are lost on import from Firefox

### DISCUSSION THREAD

**Discussion by jbcodemonkey** [dani-garcia/vaultwarden#5333] [Ideas]
Firefox exports date created, modified, and last used.
These are lost when importing to both Bitrwarden and Vaultwarden.

According to [this post](https://community.bitwarden.com/t/when-importing-vault-items-set-their-created-and-or-updated-dates-based-on-the-imported-data/48542), it seems the dates are lost even during imports from Vault/Bitwarden exports.

This information is extremely valuable to sort and determine which saved credentials are still relevant or when determining which may be newer. It would so excellent to preserve this as part of export/import operations.

**Comment by BlackDex** [comment]
That isn't something we can fix I think. If that data isn't preserved during the conversion of the import to data sent to the server then we have nothing to use.

But I'm not sure from the top of my head if that data is there. I think it wasn't in the past, but might be in newer client versions though. If not, there is nothing we can do.

**Reply by jbcodemonkey** [reply]
I am importing via the web interface from a flat file, which definitely has the fields. The issue is (I think) in how the server is processing the columns during the import. Presently, last modified and last used are set the date and time of the import.

**Reply by stefan0xC** [reply]
You can check the network request what fields are transmitted. Besides the [lastKnowRevisionDate](https://github.com/dani-garcia/vaultwarden/blob/d9b043d32ccd5ae25ef7db99c042fd0adc9b4a9f/src/api/core/ciphers.rs#L261-L267) the other fields are not sent by the client to the server.

**Reply by BlackDex** [reply]
Also, the server doesn't convert the input file or text. 
The client handles everything and send a list of regular cipher objects to the server, which are fully encrypted.
